gcp/app-engine/slack_handler.py

```diff
@@ -0,0 +1,271 @@
+    return response_text, status_code
+
+def handle_player_info_image(command_text, channel):
+    # Parse for specific player name/id and tournament name/id
+    words = command_text.split()
+    response_text = None
+    status_code = 200
+    round = None
+    if len(words) > 5:
+        if len(words) > 6:
+            player_search = words[-3]  # Assume third last word is player name/id
+            tournament_search = words[-2]  # Assume second last word is the tournament name/id
+            round = try_cast_to_int(words[-1], None)  # Assume the last word is the round
+            if round is None or round < 1 or round > 4:
+                response_text = 'When calling "player info image <player> <tournament> <round>", the round must be an integer between 1 and 4'
+                status_code = 400
+        else:
+            player_search = words[-2]  # Assume second last word is player name/id
+            tournament_search = words[-1]  # Assume the last word is the tournament name/id
+
+        if not response_text:
+            player_info = get_player(player_search, tournament_search, round)
+            if player_info:
+                try:
+                    # Create and save the image
+                    text = f"*{player_info['name']}* in *{player_info['tournament_name']}* (Round {player_info['round']}):"
+                    image = create_golf_scorecard_image(player_info)
+                    response_text, status_code = save_and_upload_slack_image(image, channel, text=text)
+                except Exception as e:
+                    response_text = f'Error creating image: {e}'
+                    status_code = 500
+            else:
+                response_text = "Player or tournament not found."
+                status_code = 404
+    else:
+        response_text = "Please specify both a player and a tournament."
+        status_code = 400
+
+    if response_text:
+        slack_client.chat_postMessage(channel=channel, text=response_text)
+    return response_text, status_code
+
+def handle_top_10_players(command_text, channel):
+    words = command_text.split()
+    status_code = 200
+    if len(words) > 3:
+        tournament_search = words[-1]  # Assume the last word is the tournament name/id
+        top_10_players = get_top_players(tournament_search, limit=10)
+        if top_10_players:
+            response_text = "*Top 10 Players:*\n"
+            for player in top_10_players:
+                response_text += (
+                    f"{player['rank']}. {player['name']} - Score: {player['score']}\n"
+                )
+        else:
+            response_text = "Tournament not found."
+            status_code = 404
+    else:
+        response_text = "Please specify a tournament."
+        status_code = 400
+
+    slack_client.chat_postMessage(
+        channel=channel,
+        text=response_text
+    )
+    return response_text, status_code
+
+def format_scores_grid(pars, scores):
```

Comment:
print a text version of the scorecard, e.g.
```
 Hole  | 1  | 2  | 3  | 4  | 5  | 6  | 7  | 8  | 9  | 10 | 11 | 12 | 13 | 14 | 15 | 16 | 17 | 18 |
-------|----|----|----|----|----|----|----|----|----|----|----|----|----|----|----|----|----|----|
 Par   | 4  | 4  | 3  | 4  | 5  | 4  | 3  | 4  | 4  | 4  | 4  | 3  | 4  | 4  | 5  | 3  | 4  | 4  |
-------|----|----|----|----|----|----|----|----|----|----|----|----|----|----|----|----|----|----|
 Score | 4  | 4  | 3  | 5  | 4  | 4  | 2  | 4  | 4  | 4  | 4  | 2  | 3  | 4  | 3  | 3  | 4  | 4  |
```

Reply:
still need to come up with a better pattern for the last row to be aligned with the rest of the grid, but will fix that in another MR